porch/pkg/git/git.go

```diff
@@ -382,6 +382,14 @@ func (r *gitRepository) DeletePackageRevision(ctx context.Context, old repositor
 		// Delete the tag
 		refSpecs.AddRefToDelete(ref)
 
+		// If this revision was proposed for deletion, we need to delete the associated branch.
```

Comment:
So this change does not update the packagerevision on the main branch right? I think doing that is a separate change does make sense.

Reply:
Yeah, it doesn't make any changes to the main branch. At some point I'd like to see if we can modify the main branch when the latest package revision is deleted, but agree that that belongs in its own PR.